src/menus/TrackerMenu/TrackerMenu.js

```diff
@@ -58,44 +59,59 @@ class TrackerMenu extends Component {
     if (layers !== prevProps.layers) {
       this.initializeClick();
     }
+
+    if (layer !== prevState.layer || feature !== prevState.feature) {
+      if (layer && feature) {
+        this.getStopSequence();
+      }
+    }
   }
 
   componentWillUnmount() {
     unByKey(this.olEventsKeys);
     this.olEventsKeys = [];
-
     this.trackerLayers.forEach((layer) => {
       layer.unClick(this.onLayerClick);
     });
   }
 
-  onLayerClick([feature], layer) {
+  onLayerClick(features, layer) {
     const { dispatchSetMenuOpen } = this.props;
-    if (feature) {
+
+    if (features.length) {
       dispatchSetMenuOpen(false);
-      layer.api
-        .fetchTrajectoryStations(
-          layer.getParams({
-            id: feature.get('id'),
-            time: getUTCTimeString(new Date()),
-          }),
-        )
-        .then((trajStations) => {
-          this.setState({
-            open: !!feature,
-            collapsed: false,
-            trajectory: { ...feature.getProperties(), ...trajStations },
-          });
-        });
-    } else {
       this.setState({
-        open: !!feature,
+        open: true,
```

Comment:
When a trajectory is deselected via map click the tracker menu stays open. I think it should close on deselect. The same bug (or feature? ;)) occurs in mobility-portal

Reply:
Make sense of course. I put back the original line

Reply:
```suggestion
        open: !!feature.length,
```

Reply:
```suggestion
        open: !!features.length,
```